src/api/providers/cerebras.ts

```diff
@@ -0,0 +1,329 @@
+import { Anthropic } from "@anthropic-ai/sdk"
+
+import { type CerebrasModelId, cerebrasDefaultModelId, cerebrasModels } from "@roo-code/types"
+
+import type { ApiHandlerOptions } from "../../shared/api"
+import { calculateApiCostOpenAI } from "../../shared/cost"
+import { ApiStream } from "../transform/stream"
+import { convertToOpenAiMessages } from "../transform/openai-format"
+import { XmlMatcher } from "../../utils/xml-matcher"
+
+import type { ApiHandlerCreateMessageMetadata, SingleCompletionHandler } from "../index"
+import { BaseProvider } from "./base-provider"
+
+const CEREBRAS_BASE_URL = "https://api.cerebras.ai/v1"
+const CEREBRAS_DEFAULT_TEMPERATURE = 0
+
+/**
+ * Removes thinking tokens from text to prevent model confusion when processing conversation history.
+ * This is crucial because models can get confused by their own thinking tokens in input.
+ */
+function stripThinkingTokens(text: string): string {
+	// Remove <think>...</think> blocks entirely, including nested ones
+	return text.replace(/<think>[\s\S]*?<\/think>/g, "").trim()
+}
```

Comment:
Some of the other providers have code like this to parse out the thinking tokens and display it in the reasoning block visually - might be a slightly nicer approach if it works?

```
			const matcher = new XmlMatcher(
				"think",
				(chunk) =>
					({
						type: chunk.matched ? "reasoning" : "text",
						text: chunk.data,
					}) as const,
			)
```

Reply:
Oh, I see that you have it below! Is it still necessary to strip the tokens then?